app/src/main/java/com/compscieddy/meetinthemiddle/ui/StatusAdapter.java

```diff
@@ -0,0 +1,154 @@
+package com.compscieddy.meetinthemiddle.ui;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.compscieddy.meetinthemiddle.R;
+
+import de.hdodenhof.circleimageview.CircleImageView;
+
+/**
+ * Created by Darren on 11-Jun-16.
+ */
+public class StatusAdapter extends RecyclerView.Adapter<StatusAdapter.StatusHolder> {
+
+  private Context mContext;
+  private static ClickListener mClickListener;
+
+  private String[] mDataSet;
+  private int[] mDataSetTypes;
+
+  public static final int CASINO = 0;
+  public static final int FITNESS = 1;
+  public static final int MOVIES = 2;
+  public static final int SLEEPING = 3;
+
+  public interface ClickListener {
+    void OnItemClick(View v);
+  }
+
+  public void setClickListener(ClickListener clickListener) {
+    mClickListener = clickListener;
+  }
+
+  public static class StatusHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```

Comment:
You should only have one StatusHolder - this is reused for "CasinoHolder", "FitnessHolder", etc. Basically delete all the other holders except for StatusHolder. You should have 1 holder per layout, and we only need 1 layout because these status buttons all look the same; you'll just be swapping out the android:src icon